docs/Developers.md

```diff
@@ -0,0 +1,119 @@
+# SRT Developer's Guide
+
+* [Development Setup](#development-setup)
+  * [Installing Dependencies](#installing-dependencies)
+  * [Forking SRT on GitHub](#forking-srt-on-github)
+  * [Building SRT](#building-srt)
+* [Project Structure](#project-structure)
+* [Coding Rules](#rules)
+* [Submitting an Issue](#submitting-an-issue)
+* [Submitting a Pull Request](#submitting-a-pull-request)
+  * [Commit Message Format](#commit-message-format)
+
+## Development Setup
+
+This document describes how to set up your development environment to build and test SRT, and explains the basic mechanics of using `git` and `cmake`.
+
+### Installing Dependencies
+
+Install the following dependencies on your machine:
+
+* [Git](http://git-scm.com/): The [Github Guide to
+  Set Up Git][git-setup] is a good source of information.
+
+* [CMake](http://cmake.org): v2.8.12 or higher is recommended.
+
+* [OpenSSL](http://www.openssl.org): v1.1.0 or higher. Alternatively Nettle and mbedTLS can be used. 
+
+### Forking SRT on Github
+
+To contribute code to SRT, you must have a GitHub account so you can push code to your own
+fork of SRT and open Pull Requests in the [GitHub Repository][github].
+
+To create a Github account, follow the instructions [here](https://github.com/signup/free).
+Afterwards, go ahead and [fork](http://help.github.com/forking) the
+[main SRT repository][github].
+
+### Building SRT
+
+To build SRT, clone the source code repository and use CMake to generate system-dependent build files:
+
+```shell
+# Clone your Github repository to 'srt' folder.
+git clone https://github.com/<github username>/srt.git srt
+
+# Go to the SRT directory.
+cd srt
+
+# Add the main SRT repository as an upstream remote to your repository.
+git remote add upstream "https://github.com/Haivision/srt.git"
+
+# For MacOS also export OpenSSL paths with the following commands:
+# export OPENSSL_ROOT_DIR=$(brew --prefix openssl)
+# export OPENSSL_LIB_DIR=$(brew --prefix openssl)"/lib"
+# export OPENSSL_INCLUDE_DIR=$(brew --prefix openssl)"/include"
+
+# Create a directory for build artifacts.
+# Note: To create a directory on windows use `md` command instead of `mkdir`.
+mkdir _build && cd _build
+
+# Generate build files, including unit tests.
+cmake .. -DENABLE_UNITTESTS=ON
+
+# Build SRT.
+cmake --build ./
+```
+
+**Note.** If you're using Windows, please refer to [Windows Build Instructions](./build-win.md).
+
+**Note.** Please see the following document for the build options: [BuildOptions.md](./BuildOptions.md). To see the full list of make options run `cmake .. -LAH` from the `_build` folder. 
+
```

Comment:
```suggestion
**Note.** Alternatively on UNIX-based systems `./configure` script can be used instead of CMake. Please see the following document for the build options: [BuildOptions.md](./BuildOptions.md). To see the full list of make options run `cmake .. -LAH` from the `_build` folder. 

```

Reply:
I'd say something more here.

```
The `configure` script provided here is **NOT** an alternative Autotools build, but a convenience script. It processes the usual format of `--long-options` and calls `cmake` with appropriate options in the end. This script is dependent on "tcl" package.
```